src/lib-components/layout/DateFilter.vue

```diff
@@ -21,12 +21,16 @@ const emits = defineEmits<{
   (e: "date-range-changed", val: DateRange): void
 }>()
 
-const sortDirChanged = (sortDir: SortDir) => {
-  emits("sort-dir-changed", sortDir)
+const sortDirChanged = (sortDir: unknown) => {
+  if (sortDir === "asc" || sortDir === "desc") {
+    emits("sort-dir-changed", sortDir)
+  }
 }
 
-const dateRangeChanged = (dateRange: DateRange) => {
-  emits("date-range-changed", dateRange)
+const dateRangeChanged = (dateRange: DateRange | undefined) => {
+  if (dateRange) {
+    emits("date-range-changed", dateRange)
+  }
```

Comment:
We didn't explicitly type the `update:model-value` event in our input components, so they've all been resolving to `any`.  `defineModel` is now inferring the type of the value in the callback based on the potential initial values, which will often include `null` and `undefined` depending on the input component.  A little type guarding will be necessary where we bind to `@update:model-value`.  I may be able to improve the type narrowing with generics in the future, but it's not terribly common at the moment.